libdrizzle/field.cc

```diff
@@ -402,9 +402,18 @@ drizzle_return_t drizzle_state_binary_field_read(drizzle_st *con)
 {
   drizzle_return_t ret;
 
-  switch(con->result->column_buffer[con->result->field_current].type)
+  /* Skip NULL fields */
+  while (con->result->null_bitmap[(con->result->field_current_read + 2) / 8] &
+         (1 << ((con->result->field_current_read + 2) % 8)))
+  {
+    con->result->field_current_read++;
+  }
+
+  switch(con->result->column_buffer[con->result->field_current_read].type)
   {
     case DRIZZLE_COLUMN_TYPE_NULL:
+      /* This should not happen since NULL fields are masked by `null_bitmap`, but cover it anyway */
```

Comment:
This might be better since it won't halt the program and will allow the user to handle the error and get an error message 

```cpp
drizzle_return_t drizzle_state_binary_field_read(drizzle_st *con)
...
  /* Skip NULL fields */
  while (con->result->null_bitmap[(con->result->field_current_read + 2) / 8] &
         (1 << ((con->result->field_current_read + 2) % 8)))
  {
    con->result->field_current_read++;
  }

  drizzle_column_type_t column_type =
    con->result->column_buffer[con->result->field_current_read].type;

  switch(column_type)
  {
    case DRIZZLE_COLUMN_TYPE_NULL:
      /* This should not happen since NULL fields are masked by `null_bitmap`,
       * but cover it anyway */
      drizzle_set_error(con, __func__,
        "Unexpected column type: DRIZZLE_COLUMN_TYPE_NULL");
      con->result->field_size= 0;
      return DRIZZLE_RETURN_UNEXPECTED_DATA;
      break;
    case DRIZZLE_COLUMN_TYPE_TINY:
...
```